src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala

```diff
@@ -75,7 +72,6 @@ class ArticlesRepository(quill: SqliteZioJdbcContext[SnakeCase], dataSource: Dat
     } yield (ar, pr, tr.map(_._2), fr.map(_._2), false))
       .map(_.headOption)
       .map(_.map(mapToArticleData))
-      .provide(dsLayer)
```

Comment:
I'm surprised that the types of the methods don't change despite the missing .provide - how does that work?

Reply:
Creating SQliteZioJdbcContext[SnakeCase] has return type ZLayer[Any, Nothing, SQliteZioJdbcContext[SnakeCase]]. Method run that SQliteZioJdbcContext[SnakeCase] provides has result type ZIO[DataSource, SQLException, T]. Creating Quill.Sqlite[SnakeCase] has result type ZLayer[DataSource, Nothing, Quill.Sqlite[SnakeCase]], however its run method returns ZIO[Any, SQLException, T]. In short it captures DataSource from the outer scope when creating quill context, and because of that DataSource does not need to be explicitly provided for each query inside the service.

Reply:
ah the `run` is different. Ok, clear then, thanks :)

Reply:
so the `run` is also the transaction boundary?

Reply:
Based on the return type and what I've read, yes

Reply:
Ok, that's good to know :) Do any of the service methods require running multiple operations in a single transaction? Or is it fine that each repository operation starts a new tx?

Reply:
@adamw I added the issue to review transaction boundaries later. Is this a blocking issue? If not, could you approve?